cmd/source/cm-watcher/main.go

```diff
@@ -0,0 +1,106 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/hashicorp/go-plugin"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+	watchtools "k8s.io/client-go/tools/watch"
+
+	"github.com/kubeshop/botkube/pkg/api/source"
+)
+
+const pluginName = "cm-watcher"
+
+type (
+	// Config holds executor configuration.
+	Config struct {
+		ConfigMap Object
+	}
+	// Object holds information about object to watch.
+	Object struct {
+		Name      string          `yaml:"name"`
+		Namespace string          `yaml:"namespace"`
+		Event     watch.EventType `yaml:"event"`
+	}
+)
+
+// CMWatcher implements Botkube source plugin.
+type CMWatcher struct{}
```

Comment:
Code style consistency - `type` block above combined with single line declaration.

Reply:
Actually, So I use the `type` block to "group" related entities, so the code block about groups types related to the configuration. This one is about the "service" that implements the source plugin functionality. 

Are you OK with such approach?